pymongo/synchronous/collection.py

```diff
@@ -231,6 +231,12 @@ def __init__(
         )
         if not isinstance(name, str):
             raise TypeError("name must be an instance of str")
+        from pymongo.synchronous.database import Database
+
+        if not isinstance(database, Database):
```

Comment:
I'm paranoid this might break apps that Mock pymongo classes but it should be fine. Anything that's mocking should inherit from our base classes anyway (instead of trying to ducktype).

Reply:
Good point, we'll have to watch out for bug reports with this.